src/ui/components/Rating/index.js

```diff
@@ -49,8 +49,27 @@ export class RatingBase extends React.Component {
     this.props.onSelectRating(rating);
   }
 
+  // Helper function used to render title attributes
+  // for each individual star, as well as the wrapper
+  // that surrounds the read-only set of stars
+  renderTitle = (i18n, rating, readOnly, starRating) => {
+    if (readOnly) {
+      if (rating) {
+        return i18n.sprintf(i18n.gettext('Rated %(rating)s out of 5.'),
+          { rating: i18n.formatNumber(parseFloat(rating).toFixed(1)) });
+      }
+      return i18n.gettext('This add-on has not been rated yet.');
+    }
+    if (rating) {
+      return i18n.sprintf(i18n.gettext(
+        `Update your rating to %(starRating)s out of 5.`), { starRating });
+    }
+    return i18n.sprintf(i18n.gettext(
+      `Rate this add-on %(starRating)s out of 5.`), { starRating });
+  }
+
   renderRatings() {
-    const { readOnly } = this.props;
+    const { i18n, readOnly } = this.props;
```

Comment:
Oops, sorry I missed this last time, but I don't think you need to accept extract `i18n` from `this.props` here, only to pass it into `this.renderTitle`. You could just add a `const { i18n } = this.props;` at the top of `renderTitle`, I believe.

Reply:
I went ahead and implemented it but it broke about 8 tests (which is probably too large to paste here). Should I go ahead and refactor them?